iroh-net/src/key.rs

```diff
@@ -227,7 +227,11 @@ impl Debug for PublicKey {
 
 impl Display for PublicKey {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", base32::fmt(self.as_bytes()))
+        if f.alternate() {
```

Comment:
when/how is this triggered?

Reply:
shouldn’t it be the other way around then,? # prints longer more detailed messages

Reply:
I could be rather tempted to switch this around and do the short format by default and the long format for alternate.  That would reduce rather a lot of verbosity in our code.

Reply:
the issue is that, even externally, we rely on `to_string()` roundtriping

Reply:
yeah, it'd probably break things if this can't be round-tripped.  so maybe we'll have to settle for this.